doc/design_docs/alternate_enum_cases.md

```diff
@@ -0,0 +1,323 @@
+
+#### Pros
+
+  * Flexible and straightforward for users
+
+#### Cons
+
+  * Adds complexity to the grammar and front-end.
+    * Not as simple of an implementation as it first appears.
+  * Allows Emboss definition files to diverge from each other, which goes
+    against the design goals of Emboss where all .emb files should look similar
+    to each other.
+    * Additionally it adds cognitive overhead in reading an unfamiliar Emboss
+      definition in a different "style".
+  * Backend/language considerations.
+    * A style used in C++ (`kCamelCase`) would also be used in languages where
+      that is not the style.
+    * Setting the name for all languages could cause issues in languages where
+      the case of a variable has semantic meaning, like the visibility of a
+      variable in `Go`.
+
+### Specifying An Exact Name In Attributes
+
+Instead of specifying a case transformation in an attribute, provide the
+specific name to be emitted. For example:
+
+```
+enum Foo:
+  BAR             = 1  [(cpp) name: "kBar"]
+  BAZ             = 2  [(cpp) name: "kBaz"]
+  MULTI_WORD_ENUM = 4  [(cpp) name: "kMultiWordEnum"]
+```
+
+Note that the proposed `enum_case` design does not preclude an attribute of this
+nature for resolving other use-cases. Either multiple names could be emitted
+(those generated by `enum_case` as well as `name`), or under the principle of
+"specific overrides general" a `name`-like attribute could override any
+`enum_case` attribute. The specifics of this should be addressed in a design for
+that feature. Additionally, the implementation of `enum_case` should be made to
+allow `name` or a similar attribute to be added without major refactoring.
+
+#### Pros
+
+  * Simple to implement
+  * Applies to more than just `enum` fields.
+  * Applies to other use cases (working around restrictions/reserved keywords in
+    backends that are not also restricted/reserved in Emboss).
+
+#### Cons
+
+  * Not possible to provide a `$default` attribute that applies generically to
+    all enum fields.
+    * This would require an attribute added to every enum member if the intent
+      is to always use a particular style.
+    * Requires a user to specify the translation for every field, making it
+      easier to mix cases or styles unintentionally.
+      * If mixing cases is intended, this is still possible with the `enum_case`
+	    attribute by overriding the default.
+
+### Transitional Cases or Attributes
+
+This alternative design would still use `enum_case` or something similar, but
+not allow multiple case options to be asserted. Instead, either a new
+transition-specific case or a transitional attribute would be used to mark a
+transition in progress. For example:
+
+```
+enum Foo:
+  BAR             = 1  [(cpp) enum_case: "kCamelCase-transitional"]
+  BAZ             = 2  [(cpp) enum_case: "kCamelCase-transitional"]
+  MULTI_WORD_ENUM = 4  [(cpp) enum_case: "kCamelCase-transitional"]
```

Comment:
This seems easy to combine with your preferred design, by putting a `-deprecated` tag on the deprecated option(s) instead of a `-transitional` tag on the "not deprecated" option:

```
enum Foo:
  BAR_BAZ = 1
    [(cpp) enum_case: "SHOUTY_CASE -deprecated, CamelCase -deprecated, kCamelCase"]
```

(This would not need to be part of the initial implementation, so you could leave it as a "future work" item.)

Reply:
Ah, that's a great idea! Added to the future work section.